tools/gfx/render.h

```diff
@@ -1085,15 +1085,32 @@ struct Viewport
 class IRenderer: public ISlangUnknown
 {
 public:
+    struct SlangDesc
+    {
+        slang::IGlobalSession* slangGlobalSession = nullptr; // (optional) A slang global session object. If null will create automatically.
+
+        SlangMatrixLayoutMode defaultMatrixLayoutMode = SLANG_MATRIX_LAYOUT_ROW_MAJOR;
+
+        char const* const* searchPaths = nullptr;
+        SlangInt            searchPathCount = 0;
+
+        slang::PreprocessorMacroDesc const* preprocessorMacros = nullptr;
+        SlangInt                        preprocessorMacroCount = 0;
+
+        const char* targetProfile = nullptr; // (optional) Target shader profile. If null this will be set to platform dependent default.
+        SlangFloatingPointMode floatingPointMode = SLANG_FLOATING_POINT_MODE_DEFAULT;
+        SlangOptimizationLevel optimizationLevel = SLANG_OPTIMIZATION_LEVEL_DEFAULT;
+    };
```

Comment:
Could this just use `slang::SessionDesc` instead? Some of the fields there might need to be ignored, but it seems like a safer way to make the `IRenderer` API independent of the details of the Slang API.

This would be especially important if/when we start to make APIs like `creaseSession` "extensible" in a binary-compatible fashion; we'd want to ensure that the `gfx` API doesn't need any changes to expose new facilities that are part of Slang session creation.